notebook/static/base/js/dialog.js

```diff
@@ -86,7 +86,7 @@ define(function(require) {
             var button = $("<button/>")
                 .addClass("btn btn-default btn-sm")
                 .attr("data-dismiss", "modal")
-                .text(label);
+                .text(i18n.translate(label).fetch());
```

Comment:
Yeah, it's kind of confusing until you understand the dialog.js code, but it does work properly, as long as you have the button labels defined properly.